Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -5224,27 +5224,27 @@ static BOOL shouldUseHighlightsForMarkedText(NSAttributedString *string)
     LOG(TextInput, "setMarkedText:\"%@\" selectedRange:(%u, %u) replacementRange:(%u, %u)", string, selectedRange.location, selectedRange.length, replacementRange.location, replacementRange.length);
 
 #if HAVE(INLINE_PREDICTIONS)
-    BOOL hasTextCompletion = [&] {
-        if (!isAttributedString)
-            return NO;
+    if (isAttributedString) {
+        NSAttributedString *attributedString = (NSAttributedString *)string;
```

Comment:
Could be rewritten to have variable assignment inside the if. And use `dynamic_objc_cast`.

Reply:
wdym?

Reply:
```
if (RetainPtr attributedString = dynamic_objc_cast<NSAttributedString>(string)) {
    …
}
```

Reply:
isn't that technically resulting in an extra call to `isKindOfClass`? (since the existing code still needs the `BOOL isAttributedString = [string isKindOfClass:[NSAttributedString class]];` to exist. I guess it doesn't really matter though